### ISSUE TITLE
Feature Request: Ability for Mergemods to update entire (non-vanilla) classes

### ISSUE THREAD

**Issue by Tajfun403** [ME3Tweaks/MassEffectModManager#381]
**The issue:**
As of this moment, performing an update to a class which was added to a startup file by a mod is not perfectly comfortable.
A new custom class can be very simply pulled into a startup file by utilizing `assetupdate` and `canmergeasnew` properties of a mergemod. However, it cannot be updated in the same easy method anymore: `assetupdate` would only update the main class export, leaving out updates committed to existing functions (because their exports already exist - so they won't be merged in again).
This proves releasing further updates to end users difficult.

**Current alternatives:**
- Asking end users to vanilla the impacted pcc files before telling them to install the mod again adds a large amount of steps that heavily complicate the mod install. It can also mean the users might need to reinstall a large amount of other mods just to update this single one properly.
- Instead of relying on `assetupdate`, new updates to every single function could be distributed as separate `addtoclassorreplace` statements. This however makes the setup required for each update significant, and in case a lot of code was modified, it could quickly escalate into a prohibitive amount of statements.

**Solutions:**
- It would be beneficial to introduce an option which would allow one to trash existing entries of a class and all its children, before merging in its content anew.
- Another solution would be to make sure all entries referenced by the merged-in assets will be pulled anew, instead of the current functionality where they are only pulled in if they were not already present in the file. This would make it additionally beneficial in case the class also supplied external new objects - such as textures - as those could be easily updated as well with a single `assetupdate`, rather than through several for each entry.
- Another another solution, as proposed by @Mgamerz, would be to allow replacement of entire classes through scriptupdate, as long as it is a non-vanilla class. As discussed over discord here, 2023-08-23: 
![rgrfg](https://github.com/ME3Tweaks/ME3TweaksModManager/assets/44444384/b46bce9d-479f-43da-8f4f-bc6485eec450)
Additionally, this option would also allow one to effectively clear a class (by scriptupdating it with an empty stub), achieving same possibilities as in the first proposed solution.

**Comment by Mgamerz** [comment]
Been looking into this. There will be a new flag on merge mods that will allow you to overwrite an existing non-vanilla class, which will be the same behavior as if you went into the Script tab in LEX and replaced the entire class text and pressed compile.

However, something I did not think about is how two classes interact, and will require some additional thought by developers.

For example, in Mass Effect 2 Randomizer LE, I have a class in SFXGame - MERControl. It references another class I also made in SFXGame - BioMorphFaceUtility. In the event that BioMorphFaceUtillity references MERControl, and I update MERControl, there is no guarantee that the changes made will be compatible. Developers will have to keep mods in sync or the game likely will become unstable.
